android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -829,6 +845,54 @@ private void finishPlayerInitialization() {
         }
     }
 
+    private static ExoMediaDrm createFrameworkMediaDrm(UUID uuid) {
+        try {
+            return VirtuosoFrameworkMediaDrm.newInstance(uuid);
+        } catch (UnsupportedDrmException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private DrmSessionManager buildDrmSessionManager(String assetId) {
+        if (assetId == null) {
+            return null;
+        }
+        Virtuoso virtuoso = new Virtuoso(getContext());
+
+        IAsset asset = null;
+
+        List<IIdentifier> list = virtuoso.getAssetManager().getByAssetId(assetId);
+        if (list != null && list.size() > 0) {
+            asset = (IAsset) list.get(0);
+        }
+
+        ISegmentedAsset segmentedAsset = asset instanceof ISegmentedAsset ? (ISegmentedAsset) asset : null;
+
+        if (segmentedAsset != null) {
+            String drmUuid = segmentedAsset.contentProtectionUuid();
+            if (drmUuid != null) {
+                UUID drmSchemeUuid = null;
+                if (!TextUtils.isEmpty(drmUuid))
+                    drmSchemeUuid = Util.getDrmUuid(drmUuid);
+
+                if (drmSchemeUuid != null) {
+                    try {
+                        ExoMediaDrm.Provider mediaDrmProvider = uuid -> createFrameworkMediaDrm(uuid);
+                        return new ExoplayerDrmSessionManager.Builder(segmentedAsset)
+                                .setUuidAndExoMediaDrmProvider(drmSchemeUuid, mediaDrmProvider).build();
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+
+            }
+        }
+
+        return null;
```

Comment:
I don't think it ever makes sense to return a null drm session manager. 100% of our content requires DRM, so anything that results in a null drm session manager would be an error.

It's far far easier dealing with and debugging code that doesn't translate every error state into a null and continue.